test/jdk/javax/swing/JList/4618767/JListSelectedElementTest.java

```diff
@@ -0,0 +1,217 @@
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+import java.awt.Robot;
+import java.awt.event.KeyEvent;
+import java.util.Arrays;
+import java.util.List;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.stream.Collectors;
+import javax.swing.JFrame;
+import javax.swing.JList;
+import javax.swing.JMenu;
+import javax.swing.JMenuBar;
+import javax.swing.JMenuItem;
+import javax.swing.SwingUtilities;
+import javax.swing.UIManager;
+import javax.swing.UnsupportedLookAndFeelException;
+import javax.swing.event.MenuEvent;
+import javax.swing.event.MenuListener;
+
+import static javax.swing.UIManager.getInstalledLookAndFeels;
+
+/*
+ * @test
+ * @key headful
+ * @bug 4618767
+ * @summary Typing a letter while a JList has focus now makes the selection jump to the item whose text
+ *          starts with that letter even though that letter is accompanied by modifier keys such as ALT or CTRL.
+ * @run main JListSelectedElementTest
+ */
+public class JListSelectedElementTest {
+
+    private static final int MENU = KeyEvent.VK_F;
+    private static JFrame frame;
+    private static JList<String> list;
+    private static Robot robot;
+    private static volatile boolean menuSelected;
+
+    public static void main(String[] args) throws Exception {
+        runTest();
+    }
+
+    public static void runTest() throws Exception {
+        robot = new Robot();
+        robot.setAutoWaitForIdle(true);
+        robot.setAutoDelay(200);
+
+        final boolean isAMac = System.getProperty("os.name").toLowerCase().contains("os x");
+
+        List<String> lafs = Arrays.stream(getInstalledLookAndFeels())
+                                  .map(UIManager.LookAndFeelInfo::getClassName)
+                                  .collect(Collectors.toList());
+        for (final String laf : lafs) {
+            try {
+                AtomicBoolean lafSetSuccess = new AtomicBoolean(false);
+                SwingUtilities.invokeAndWait(() -> {
+                    lafSetSuccess.set(setLookAndFeel(laf));
+                    if (lafSetSuccess.get()) {
+                        createUI();
+                    }
+                });
+                if (!lafSetSuccess.get()) {
+                    continue;
+                }
+                robot.waitForIdle();
+
+                // Request for focus and wait until the list has focus.
+                SwingUtilities.invokeAndWait(() -> list.requestFocusInWindow());
+                int waitCount = 0;
+                while (!isFocusOwner()) {
+                    robot.delay(100);
+                    waitCount++;
+                    if (waitCount > 20) {
+                        throw new RuntimeException("Waited for long, but can't get focus for list");
+                    }
+                }
```

Comment:
Maybe you should wait for `FocusEvent.FOCUS_GAINED` event?

In some way, requesting focus to list is redundant: it's the only component in the frame, there's no other component to get the focus.